src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash/fp'
+import { Ajax } from 'src/libs/ajax'
+import * as Utils from 'src/libs/utils'
+import { ToolLabel, tools, toolToExtensionMap } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { CloudProviderType } from 'src/pages/workspaces/workspace/workspace-utils'
+
+// This syntax gives us nominal typing
+// Nominal types are not interchangable even if the underlying type is the same
+// A func that takes (a: A, b: B) where A and B are both strings cannot be called like (b,a) with this restriction
+export type FileName = string & { __typeToken: 'FileName' }; // represents a file with an extension and no path, eg `dir/file.ipynb` =>  `file.ipynb`
+export type AbsolutePath = string & { __typeToken: 'AbsolutePath' }; // represents an absolute path in the context of a cloud storage directory structure, i.e. `dir/file.ipynb`
+export type Extension = string & { __typeToken: 'Extension' }; // represents the substring found after the last dot in a file, eg `dir/file.txt.ipynb` => `ipynb`
+export type DisplayName = string & { __typeToken: 'DisplayName' }; // represents the name of a file without an extension or pathing eg `dir/file.txt.ipynb`=> `dir/file.txt`
```

Comment:
+1 on considering extracting type definitions out into its own file in general, since I tend to refer to them more frequently than implementation code and it helps to be able to see them in one place.